modules/platform/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/filter/IdempotentRequestHandler.java

```diff
@@ -0,0 +1,168 @@
+import org.nuxeo.runtime.kv.KeyValueStore;
+import org.nuxeo.runtime.services.config.ConfigurationService;
+
+/**
+ * Handles idempotency key in POST requests.
+ * <p>
+ * If {@link #HEADER_KEY} is found in the request header, will intercept request handling to:
+ * <ul>
+ * <li>mark the request as being processed
+ * <li>capture the response when request was processed without any error and store it
+ * <li>return the stored response if a subsequent request with the same key is processed again
+ * <li>return a conflict response if a request with the same key is processed while the first request is still in
+ * progress.
+ * </ul>
+ *
+ * @since 11.5
+ */
+public class IdempotentRequestHandler {
+
+    private static final Logger log = LogManager.getLogger(IdempotentRequestHandler.class);
+
+    /** @since 11.5 */
+    public static final String HEADER_KEY = "Idempotency-Key";
+
+    /** @since 11.5 */
+    public static final String STORE_PROPERTY = "org.nuxeo.request.controller.keyvaluestore.name";
+
+    protected static final String DEFAULT_STORE = "idempotentrequest";
+
+    /** 1 day */
+    protected static final int DEFAULT_TTL_SECONDS = 86400;
+
+    /** @since 11.5 */
+    public static final String TTL_SECONDS_PROPERTY = "org.nuxeo.request.controller.ttl.seconds";
+
+    /** @since 11.5 */
+    public static final String INPROGRESS_MARKER = "IDEMPOTENCY_INPROGRESS_MARKER";
+
+    /** @since 11.5 */
+    public static final String STATUS_SUFFIX = "_status";
+
+    // safe methods according to RFC 7231 4.2.1
+    protected static final Set<String> SAFE_METHODS = new HashSet<>(
+            Arrays.asList(HttpGet.METHOD_NAME, HttpHead.METHOD_NAME, HttpOptions.METHOD_NAME, HttpTrace.METHOD_NAME));
+
+    // idempotent methods according to RFC 7231 4.2.2
+    protected static final Set<String> IDEMPOTENT_METHODS = new HashSet<>(
+            Arrays.asList(HttpPut.METHOD_NAME, HttpDelete.METHOD_NAME));
+
+    protected int getTTLSeconds() {
+        ConfigurationService cs = Framework.getService(ConfigurationService.class);
+        if (cs != null) {
+            return cs.getInteger(TTL_SECONDS_PROPERTY, DEFAULT_TTL_SECONDS);
+        }
+        return DEFAULT_TTL_SECONDS;
+    }
+
+    protected String getStoreName() {
+        ConfigurationService cs = Framework.getService(ConfigurationService.class);
+        if (cs != null) {
+            return cs.getString(STORE_PROPERTY, DEFAULT_STORE);
+        }
+        return DEFAULT_STORE;
+    }
+
+    protected boolean shouldIgnore(String method) {
+        return SAFE_METHODS.contains(method) || IDEMPOTENT_METHODS.contains(method);
+    }
+
+    public void handle(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        String method = request.getMethod();
+        log.debug("Method: " + method);
+        if (shouldIgnore(method)) {
+            log.debug("Request ignored");
```

Comment:
Add detail: `No idempotent processing done: method is already idempotent` (the request isn't really ignored)
I would actually log the method here and not in a separate line.